### ISSUE TITLE
kops rolling-update fails if node to cordon not found

### ISSUE THREAD

**Issue by heybronson** [kubernetes/kops#13149]
/kind bug

**1. What `kops` version are you running? The command `kops version`, will display
 this information.**
`v1.24.0-alpha.2`

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**
`1.20.9`

**3. What cloud provider are you using?**
AWS

**4. What commands did you run?  What is the simplest way to reproduce this issue?**
`kops rolling-update cluster --yes`

**5. What happened after the commands executed?**
`kops rolling-update` exited when it attempted to cordon a node that didn't exist.
```
I0125 01:19:32.210359    9433 instancegroups.go:310] Tainting 1 node in "spot-c-us-west-2b" instancegroup.

I0125 01:19:32.214793    9433 instancegroups.go:567] Detaching instance "i-0f672b60a087493a2", node "ip-10-4-176-60.us-west-2.compute.internal", in group "<REDACTED>".

I0125 01:19:32.641192    9433 instancegroups.go:171] waiting for 15s after detaching instance

I0125 01:19:47.643334    9433 instancegroups.go:469] Validating the cluster.

I0125 01:19:55.870258    9433 instancegroups.go:505] Cluster validated; revalidating in 10s to make sure it does not flap.

I0125 01:20:13.265808    9433 instancegroups.go:502] Cluster validated.

I0125 01:20:13.265855    9433 instancegroups.go:399] Draining the node: "ip-10-4-176-60.us-west-2.compute.internal".

I0125 01:20:13.270149    9433 instancegroups.go:469] Validating the cluster.

I0125 01:20:19.230930    9433 instancegroups.go:502] Cluster validated.

I0125 01:20:19.230955    9433 instancegroups.go:310] Tainting 1 node in "spot-c-us-west-2c" instancegroup.

I0125 01:20:19.242305    9433 instancegroups.go:567] Detaching instance "i-0d82c4d6be3a586c2", node "ip-10-4-209-56.us-west-2.compute.internal", in group "<REDACTED>".

I0125 01:20:19.588839    9433 instancegroups.go:171] waiting for 15s after detaching instance

I0125 01:20:34.593420    9433 instancegroups.go:469] Validating the cluster.

I0125 01:20:41.731842    9433 instancegroups.go:525] Cluster did not pass validation, will retry in "30s": machine "i-0a2614f66b3fca27a" has not yet joined cluster.

I0125 01:21:18.689067    9433 instancegroups.go:525] Cluster did not pass validation, will retry in "30s": machine "i-0a2614f66b3fca27a" has not yet joined cluster.

I0125 01:21:56.977687    9433 instancegroups.go:525] Cluster did not pass validation, will retry in "30s": node "ip-10-4-206-226.us-west-2.compute.internal" of role "node" is not ready, <REDACTED> is pending.

I0125 01:22:34.554014    9433 instancegroups.go:525] Cluster did not pass validation, will retry in "30s": node "ip-10-4-206-226.us-west-2.compute.internal" of role "node" is not ready, <REDACTED> is pending.

I0125 01:23:11.679642    9433 instancegroups.go:525] Cluster did not pass validation, will retry in "30s": system-cluster-critical pod <REDACTED> is pending.

I0125 01:23:50.401230    9433 instancegroups.go:525] Cluster did not pass validation, will retry in "30s": system-cluster-critical pod <REDACTED> is not ready.

I0125 01:24:26.479995    9433 instancegroups.go:505] Cluster validated; revalidating in 10s to make sure it does not flap.

I0125 01:24:42.641906    9433 instancegroups.go:502] Cluster validated.

I0125 01:24:42.641956    9433 instancegroups.go:399] Draining the node: "ip-10-4-209-56.us-west-2.compute.internal".

WARNING: ignoring DaemonSet-managed Pods: <REDACTED>

evicting pod <REDACTED>

I0125 01:24:44.325156    9433 request.go:665] Waited for 1.000557774s due to client-side throttling, not priority and fairness, request: POST:<REDACTED>

I0125 01:25:26.355238    9433 instancegroups.go:657] Waiting for 5s for pods to stabilize after draining.

I0125 01:25:31.356192    9433 instancegroups.go:418] deleting node "ip-10-4-209-56.us-west-2.compute.internal" from kubernetes

I0125 01:25:31.364704    9433 instancegroups.go:590] Stopping instance "i-0d82c4d6be3a586c2", node "ip-10-4-209-56.us-west-2.compute.internal", in group "<REDACTED>" (this may take a while).

I0125 01:25:31.683100    9433 instancegroups.go:436] waiting for 15s after terminating instance

I0125 01:25:46.684875    9433 instancegroups.go:469] Validating the cluster.

I0125 01:25:53.120781    9433 instancegroups.go:505] Cluster validated; revalidating in 10s to make sure it does not flap.

I0125 01:26:09.194879    9433 instancegroups.go:502] Cluster validated.

Error: failed to drain node "ip-10-4-176-60.us-west-2.compute.internal": error cordoning node: cordon error: nodes "ip-10-4-176-60.us-west-2.compute.internal" not found
```

**6. What did you expect to happen?**
`kops rolling-update` should silently fail if a node isn't found and continue to to cordon/drain other nodes.

**Comment by danports** [comment]
I've run into this too - it seems to be fairly common since cluster-autoscaler keeps doing its job during a rolling update.

**Comment by olemarkus** [comment]
Still need to work for the linked fix to be released. And cherry picked.

**Comment by heybronson** [comment]
This will be fixed in the next 1.23 or 1.24 release. Thanks
